File: app/src/main/java/com/holokenmod/GridCellBorders.java
package com.holokenmod;

import java.util.EnumMap;

public class GridCellBorders {
	private final EnumMap<Direction, GridBorderType> borders = new EnumMap<>(Direction.class);
	
	public GridCellBorders() {
		this(GridBorderType.BORDER_NONE,
				GridBorderType.BORDER_NONE,
				GridBorderType.BORDER_NONE,
				GridBorderType.BORDER_NONE);
	}
	
	public GridCellBorders(final GridBorderType north, final GridBorderType east, final GridBorderType south, final GridBorderType west) {
		borders.put(Direction.NORTH, north);
		borders.put(Direction.WEST, west);
		borders.put(Direction.SOUTH, south);
		borders.put(Direction.EAST, east);
	}
	
	public GridBorderType getBorderType(final Direction direction) {
		return borders.get(direction);
	}
	
	void setBorderType(final Direction direction, final GridBorderType borderType) {
		borders.put(direction, borderType);
	}
}

File: holoken-core/src/main/kotlin/com/holokenmod/grid/GridCellBorders.kt
package com.holokenmod.grid

import com.holokenmod.Direction

class GridCellBorders @JvmOverloads constructor(
    north: GridBorderType = GridBorderType.BORDER_NONE,
    east: GridBorderType = GridBorderType.BORDER_NONE,
    south: GridBorderType = GridBorderType.BORDER_NONE,
    west: GridBorderType = GridBorderType.BORDER_NONE
) {
    private var borders: MutableMap<Direction, GridBorderType> = mutableMapOf(
        Direction.NORTH to north,
                Direction.WEST to west,
                Direction.SOUTH to south,
                Direction.EAST to east,
    )

    fun getBorderType(direction: Direction): GridBorderType {
        return borders[direction]!!
    }

    fun setBorderType(direction: Direction, borderType: GridBorderType) {
        borders[direction] = borderType
    }
}